File: frontend/modules/transaction/types/TransactionDetails.ts
import {TransactionEventDetails} from "@transaction/types/TransactionEventDetails.ts";

export interface TransactionDetails {
    uuid: string;
    userId: string;
    quoteId: string;
    paymentId: string;
    type: string;
    status: "PENDING" | "COMPLETED" | "ERROR" | "CANCELLED" | string;
    subStatus: string;
    debitTotal: number;
    debitType: string;
    debitSubtype: string;
    debitCurrency: string,
    creditTotal: number;
    creditType: string;
    creditSubtype: string;
    creditCurrency: string;
    creditReferenceId: string;
    createdDate: string;
    updatedDate: string;
    events?: TransactionEventDetails[]
}

File: backend/modules/transaction/src/main/java/bitecode/modules/transaction/model/data/TransactionDetails.java
package bitecode.modules.transaction.model.data;

import bitecode.modules._common.shared.transaction.model.enums.TransactionStatus;
import bitecode.modules._common.shared.transaction.model.enums.TransactionSubstatus;
import bitecode.modules._common.shared.transaction.model.enums.TransactionType;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

public record TransactionDetails(
        String uuid,
        String userId,
        String paymentId,
        TransactionType type,
        TransactionStatus status,
        TransactionSubstatus subStatus,
        BigDecimal debitTotal,
        String debitType,
        String debitSubtype,
        String debitCurrency,
        BigDecimal creditTotal,
        String creditType,
        String creditSubtype,
        String creditCurrency,
        Instant createdDate,
        Instant updatedDate,
        List<TransactionEventDetails> events
) {

}
